home/lncm/bitcoin/bitcoin.conf

```diff
@@ -1,23 +1,40 @@
+# [core]
+# Set database cache size in megabytes; machines sync faster with a larger cache. Recommend setting as high as possible based upon machine's available RAM.
+dbcache=100
+# Keep at most <n> unconnectable transactions in memory.
+maxorphantx=10
+# Keep the transaction memory pool below <n> megabytes.
+maxmempool=50
+# Reduce storage requirements by only storing most recent N MiB of block. This mode is incompatible with -txindex and -rescan. WARNING: Reverting this setting requires re-downloading the entire blockchain. (default: 0 = disable pruning blocks, 1 = allow manual pruning via RPC, greater than 550 = automatically prune blocks to stay under target size in MiB).
+prune=550
+
+# [network]
+# Bind to given address and always listen on it. Use [host]:port notation for IPv6.
+bind=0.0.0.0
+# Maintain at most N connections to peers.
+maxconnections=40
+# Tries to keep outbound traffic under the given target (in MiB per 24h), 0 = no limit.
+maxuploadtarget=5000
+# Use separate SOCKS5 proxy <ip:port> to reach peers via Tor hidden services.
+onion=127.0.0.1:9050
+
+# [rpc]
+# Accept command line and JSON-RPC commands.
 server=1
+# Accept public REST requests.
 rest=1
+# Username and hashed password for JSON-RPC connections. The field <userpw> comes in the format: <USERNAME>:<SALT>$<HASH>. RPC clients connect using rpcuser=<USERNAM
+rpcauth=lncm:salt$hash
+# Listen for JSON-RPC connections on this port
+rpcport=8332
+# Allow JSON-RPC connections from specified source. Valid for <ip> are a single IP (e.g. 1.2.3.4), a network/netmask (e.g. 1.2.3.4/255.255.255.0) or a network/CIDR (
 rpcallowip=127.0.0.1
 rpcallowip=192.168.0.1/24
 rpcallowip=192.168.1.1/24
 rpcallowip=10.0.0.0/8
 rpcallowip=172.16.0.0/12
 rpcallowip=172.18.0.0/16
 
-proxy=127.0.0.1:9050
-torcontrol=127.0.0.1:9051
```

Comment:
Tor is probably running on a different host/container which address should be specified. 

Also same-host Tor relies on `CookieAuthentication` for access, and that in turn relies on a program attempting to TorControl, running as a user belonging to `debian-tor` group.  Since they'll likely be running in different containers, An access password will likely need to be:
* generated, 
* run through `tor --hash-password PASSWORD,
	* added to `torrc` as `HashedControlPassword`,
* added here as `torpassword`

Reply:
Tor is running on the host.

But all containers are in host mode so will use the same network interface as host. We can remove all the docker subnets soon as we don't need to use it

Reply:
Okay, `localhost` access is one thing.  Filesystem access to the cookie file is another.  

Not sure how Docker "host mode" works exactly, but setting up password auth might be easier than: 1. figuring out how to set file permissions properly across containers/unix groups.

Reply:
The docker container has access to the host regardless of mode as long as you configure it correctly. I've actually got this working on master

Reply:
Just to make sure:

- [ ] `bitcoind` is able to **automatically** create Tor Hidden Service,
- [ ] `torpassword=` **is not** added to `bitcoin.conf`,
- [ ] `HiddenServiceDir` **is not** specified in `torrc`,
- [ ] `CookieAuthentication 1` **is set** in `torrc`.

?

Reply:
Torcontrol is on default localhost and port, therefore redundant and removed

proxy= is being replaced with onion= so that we have hybrid communications. With bitcoind's low traffic we could do tor-only mode too but at the expense of worse connectivity

Reply:
Bitcoind needs to be able to **control** Tor too.  And for that bitcoind, needs to be able to authenticate with Tor.

Reply:
I would like to close this issue soon,

What changes do we need to make for `tor` and `bitcoind`?

Reply:
In host mode bitcoind can access the tor on the host.

I have this working already with my archival alpine box

Reply:
Agreed, bitcoind is working with this config. I'm going to merge this now.

@meeDamian If there's still an issue with the config, please re-open or create a new issue in the future